### DISCUSSION TITLE
Make use of ESP32 Hall Effect Sensor value in rules or berry scripts

### DISCUSSION THREAD

**Discussion by gkarag** [arendst/Tasmota#13741] [Q&A] [answered]
Hello,

I use the officially released Tasmota 10.0.0 and I am trying to create a rule or write a script that would act on particular values obtained from an external Hall effect sensor. Currently the sensor value shows on the Main Menu page and in the Console, but I cannot to find a way to access it. What is the proper way of accessing the sensor data from rules or scripts?

**Comment by sfromis** [comment, marked as answer]
Just like any other sensor. Look in the published JSON payload on the .../SENSOR topic, and construct a trigger, which would be `ESP32#HallEffect`.

**Reply by gkarag** [reply]
Worked for me, thank you.

**Comment by s-hadinger** [comment]
I will look at how to trigger an internal teleperiod like event to get this data at a higher frequency, like every second

**Reply by barbudor** [reply]
Possibility to call a given sensor/driver to retreive it's json without using rules could be handy
 `tasmota.driver.get_json( driverid )` and `tasmota.sensor.get_json( sensorid )` would return the JSON from a FUNC_JSON_APPEND to the appropriate Xdrv or Xsns

**Reply by sfromis** [reply]
I'd also like calls from Berry over callbacks, gives full flexibility for polling. Could be different sensors polled differently, or a high (many times a second) polling rate for a sensor/driver known to support this.

**Reply by s-hadinger** [reply]
I'm not sure to understand what you mean

**Reply by sfromis** [reply]
What I see there is to have the Berry program call the driver JSON generation (FUNC_JSON_APPEND), instead of a teleperiod-like callback, to have the polling scheduling done using tasmota.set_timer instead.

**Reply by s-hadinger** [reply]
Got it. I was thinking of something along those lines

**Comment by s-hadinger** [comment]
Please check out #13756 it gives you the ability to query sensor values

**Comment by ozett** [comment]
I'm still playing around with berry, but unfortunately I'm not a good or experienced programmer. I'm more of a typical tasmota user: enthusiastic layman. berry makes many things possible, but functional programming is not so easy to create simple scripts (or in case of berry: advanced code) from scratch. so i would be very happy (also for all other noobs) to see an example from a professional how to bring the hall effect sensor into a variable with berry. thats maybe a good start in general and good to have a skeleton for the cook-book?

activate:
https://tasmota.github.io/docs/ESP32/#hall-sensor

and than?
```
tasmota.read_sensors()
{"Time":"2023-03-02T08:49:52","ESP32":{"Temperature":35.0,"HallEffect":68},"TempUnit":"C"}
```

edit:
seems that others had also this need and provided an simple example: https://github.com/arendst/Tasmota/discussions/13975  and another here: https://github.com/arendst/Tasmota/discussions/15640
and an advanced example https://github.com/arendst/Tasmota/discussions/18046

which makes it in the simple form for my sensor-string:

```
import json
sensors=json.load(tasmota.read_sensors())
print(sensors['ESP32']['HallEffect'])
```
